src/sources/soundsourceffmpeg.cpp

```diff
@@ -276,7 +368,7 @@ bool SoundSourceFFmpeg::readFramesToCache(unsigned int count, SINT offset) {
             // Are we on correct audio stream. Currently we are always
             // Using first audio stream but in future there should be
             // possibility to choose which to use
-            if (l_SPacket.stream_index == m_iAudioStream &&
+            if (l_SPacket.stream_index == m_pAudioStream->index &&
```

Comment:
m_pAudioStream indicates if the SoundSource has been opened sucessfully. 
This function must never be called otherwise.

I think there are similar cases for other SoundSources. We could add a debug 
assertion to be sure that we didn't miss a code path, but the actual error 
handling is done at a higher level.

On 08.09.2016 21:12, Daniel Schürmann wrote:
>
> In src/sources/soundsourceffmpeg.cpp 
> <https://github.com/mixxxdj/mixxx/pull/1003#discussion_r78070733>:
>
> > @@ -276,7 +368,7 @@ bool SoundSourceFFmpeg::readFramesToCache(unsigned int count, SINT offset) {
> >              // Are we on correct audio stream. Currently we are always
> >              // Using first audio stream but in future there should be
> >              // possibility to choose which to use
> > -            if (l_SPacket.stream_index == m_iAudioStream &&
> > +            if (l_SPacket.stream_index == m_pAudioStream->index &&
>
> can't m_pAudioStream be null here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/1003/files/14399cf62b5d5ee96a4d0d3a361af8ab165b2144#r78070733>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-auUQmYhePO6_ZoxbmABDg6h3L0dpks5qoF4cgaJpZM4J2QNu>.
>